src/core/lib/gprpp/time.h

```diff
@@ -37,8 +37,8 @@
     uint64_t now = grpc_core::Timestamp::FromTimespecRoundDown( \
                        gpr_now(GPR_CLOCK_MONOTONIC))            \
                        .milliseconds_after_process_epoch();     \
-    uint64_t prev_tsamp = prev.exchange(now);                   \
-    if (prev_tsamp == 0 || now - prev_tsamp > (n)*1000) {       \
+    if (prev == 0 || now - prev > (n)*1000) {                   \
```

Comment:
why are we not updating prev all the time ? This could have an un-desired effect of some logs not getting printed at the correct frequency

Reply:
`now - prev_tsamp` is forever a small single-digit number of millis if you always update `prev` to `now`, and call this macro in a tight loop. When have you seen this working?

Reply:
I've reverted this fix since it's unrelated to the PR. I'll land it separately.